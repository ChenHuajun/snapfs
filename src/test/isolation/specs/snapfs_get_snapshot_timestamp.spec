# The recent just created snapshot cannot be older than 3 seconds

teardown {
	select pg_remove_snapshot( ( select recent_snapshot from pg_control_snapshot() ) );
	drop table foo;
}

session "s1"
step "s1_mk_sn" {
	select 1 as "make_snapshot" from pg_make_snapshot();
	create table foo as select 'test_data';
  select (
		case
			when ( ( date_trunc( 'second', now() ) - ( select pg_get_snapshot_timestamp( ( select recent_snapshot from pg_control_snapshot() ) ) ) ) between '-1 sec'::interval and '3 sec'::interval )
			then 'passed'
			else 'Now:' || date_trunc( 'second', now() )::varchar || ' SN:' || ( select pg_get_snapshot_timestamp( ( select recent_snapshot from pg_control_snapshot() ) ) )
			end
	) as "timestamp_check"
}

permutation "s1_mk_sn"
