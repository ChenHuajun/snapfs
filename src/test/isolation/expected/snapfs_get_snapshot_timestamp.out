Parsed test spec with 1 sessions

starting permutation: s1_mk_sn
step s1_mk_sn: 
	select 1 as "make_snapshot" from pg_make_snapshot();
	create table foo as select 'test_data';
	select ( now() - ( select pg_get_snapshot_timestamp( ( select recent_snapshot from pg_control_snapshot() ) ) ) ) between '0 sec'::interval and '1 min'::interval as "timestamp_check"

make_snapshot  

1              
timestamp_check

t              
pg_remove_snapshot

               
